packages/flutter_tools/lib/src/commands/build_apk.dart

```diff
@@ -227,51 +224,43 @@ class BuildApkCommand extends FlutterCommand {
 
 Future<_ApkComponents> _findApkComponents(
   TargetPlatform platform,
-  BuildConfiguration config,
-  String enginePath,
+  BuildVariant buildVariant,
   String manifest,
-  String resources,
-  BuildVariant buildVariant
+  String resources
 ) async {
-  // TODO(devoncarew): Get the right artifacts for [buildVariant].
+  _ApkComponents components = new _ApkComponents();
+  components.manifest = new File(manifest);
+  components.resources = resources == null ? null : new Directory(resources);
 
-  List<String> artifactPaths;
-  if (enginePath != null) {
+  if (tools.isLocalEngine) {
     String abiDir = platform == TargetPlatform.android_arm ? 'armeabi-v7a' : 'x86_64';
-    artifactPaths = [
-      '$enginePath/third_party/icu/android/icudtl.dat',
-      '${config.buildDir}/gen/sky/shell/shell/classes.dex.jar',
-      '${config.buildDir}/gen/sky/shell/shell/shell/libs/$abiDir/libsky_shell.so',
-      '$enginePath/build/android/ant/chromium-debug.keystore',
+    String enginePath = tools.engineSrcPath;
+    String buildDir = tools.getEngineArtifactsDirectory(platform, buildVariant).path;
+
+    components.icuData = new File('$enginePath/third_party/icu/android/icudtl.dat');
```

Comment:
I see.  This is part of @chinmaygarde's work to make the artifacts.zip buckets look like the out/Foo/dist and vice versa.  Silly to have to encode this here.